identity_stardust/examples/create_did.rs

```diff
@@ -77,15 +83,64 @@ async fn main() -> anyhow::Result<()> {
   // Create an empty DID Document.
   // All new Stardust DID Documents initially use a placeholder DID,
   // "did:stardust:0x00000000000000000000000000000000".
-  let document: StardustDocument = StardustDocument::new();
+  let example_did = CoreDID::parse("did:example:0xffffffffffffffff").unwrap();
+  let key_did = CoreDID::parse("did:key:0xf43958394883939484848484").unwrap();
+
+  let mut document: StardustDocument = StardustDocument::new();
+
+  let keypair: KeyPair = KeyPair::new(KeyType::Ed25519).unwrap();
+  document
+    .tmp_add_verification_method(
+      document.tmp_id().clone(),
+      &keypair,
+      "#did-self",
+      MethodScope::VerificationMethod,
+    )
+    .unwrap();
+
+  let keypair: KeyPair = KeyPair::new(KeyType::Ed25519).unwrap();
+  document
+    .tmp_add_verification_method(
+      example_did.clone(),
+      &keypair,
+      "#did-foreign",
+      MethodScope::authentication(),
+    )
+    .unwrap();
+
+  document
+    .tmp_add_service(
+      document.tmp_id().clone(),
+      "#my-service",
+      "RevocationList2030",
+      identity_did::service::ServiceEndpoint::One(Url::parse("https://example.com/d2a749c").unwrap()),
+    )
+    .unwrap();
+
+  document
+    .tmp_add_service(
+      key_did.clone(),
+      "#my-foreign-service",
+      "RevocationList2030",
+      identity_did::service::ServiceEndpoint::One(Url::parse("https://example.com/f4c42e9da").unwrap()),
+    )
+    .unwrap();
```

Comment:
We should move this to a unit test to keep the example clean.

Reply:
I'd wait for the implementation of `StardustDocument` to be more mature, otherwise this requires moving almost the entire example into a unit test. The serialization transformation itself is tested already.

Reply:
I'd wait for the implementation of `StardustDocument` to be more mature, otherwise this requires moving almost the entire example into a unit test. The serialization transformation itself is tested already.

Reply:
Unless the implementation of `StardustDocument` also generalises `CoreDocument` to always take `CoreDID` for `Service` and `VerificationMethod`, this example would not work after `StardustDID` is implemented anyway, since it takes foreign DID Methods.

```rust
  let example_did = CoreDID::parse("did:example:0xffffffffffffffff").unwrap();
  let key_did = CoreDID::parse("did:key:0xf43958394883939484848484").unwrap();
```

These should just be different `StardustDID` instances.

```rust
  let did1 = CoreDID::parse("did:stardust:0xffffffffffffffffffffffffffffffff").unwrap();
  let did2 = CoreDID::parse("did:stardust:0xf4395839488393948484848484848484").unwrap();
```